server/src/main/java/org/candlepin/async/tasks/AsyncJobCleaner.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.async.tasks;
+
+import org.candlepin.async.AsyncJob;
+import org.candlepin.async.JobExecutionContext;
+import org.candlepin.async.JobExecutionException;
+import org.candlepin.model.AsyncJobStatusCurator;
+import org.candlepin.util.Util;
+
+import com.google.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+
+/**
+ * AsyncJobCleaner removes finished jobs older than yesterday, and failed
+ * jobs from 4 days ago.
+ */
+public class AsyncJobCleaner implements AsyncJob {
```

Comment:
Why change the name of the job? The 'Async' prefix is not consistent with the other job names. If you're worried about them having the same name with the old one, the old job classes will eventually get deleted before this feature branch gets merged into master.

Reply:
As this cleaner job deals with async jobs hence I add Async prefix like AsyncJobResource. However, no issue to remove the async prefix. I will update PR.

Reply:
Done.